In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
data_dic = pd.read_csv('data_dictionary.csv')

In [3]:
data_dic.head(44)

Table            Field  \
0               world_cups             Year   
1               world_cups     Host Country   
2               world_cups           Winner   
3               world_cups       Runners-Up   
4               world_cups            Third   
5               world_cups           Fourth   
6               world_cups     Goals Scored   
7               world_cups  Qualified Teams   
8               world_cups   Matches Played   
9         world_cup_groups            Group   
10        world_cup_groups             Team   
11        world_cup_groups     FIFA Ranking   
12        world_cup_squads               ID   
13        world_cup_squads             Team   
14        world_cup_squads         Position   
15        world_cup_squads           Player   
16        world_cup_squads              Age   
17        world_cup_squads             Caps   
18        world_cup_squads            Goals   
19        world_cup_squads         WC Goals   
20        world_cup_squads           League   
21        world_cup_squads             Club   
22  2022_world_cup_matches               ID   
23  2022_world_cup_matches             Year   
24  2022_world_cup_matches             Date   
25  2022_world_cup_matches            Stage   
26  2022_world_cup_matches        Home Team   
27  2022_world_cup_matches        Away Team   
28  2022_world_cup_matches        Host Team   
29       world_cup_matches               ID   
30       world_cup_matches             Year   
31       world_cup_matches             Date   
32       world_cup_matches            Stage   
33       world_cup_matches        Home Team   
34       world_cup_matches       Home Goals   
35       world_cup_matches       Away Goals   
36       world_cup_matches        Away Team   
37       world_cup_matches   Win Conditions   
38       world_cup_matches        Host Team   
39   international_matches               ID   
40   international_matches       Tournament   
41   international_matches             Date   
42   international_matches        Home Team   
43   international_matches       Home Goals   

                                          Description  
0              Year in which the World Cup took place  
1                  Country that hosted the tournament  
2                     Country that won the tournament  
3                  Country that received second place  
4                   Country that received third place  
5                  Country that received fourth place  
6      Total number of goals scored in the tournament  
7   Total number of teams that participated in the...  
8    Total number of matches played in the tournament  
9   Group for the first stage of the tournament (c...  
10                                       Country name  
11                               Current FIFA Ranking  
12                   Unique identifier for the player  
13                       Country the player plays for  
14  Position the player plays on the team (Goalkee...  
15                                        Player name  
16                                         Player age  
17              Matches played with the national team  
18                 Goals scored for the national team  
19       World Cup goals scored for the national team  
20  Country for the domestic league the player pla...  
21                 Domestic club the player plays for  
22                    Unique identifier for the match  
23        Year in which the World Cup will take place  
24                                  Data of the match  
25                            Stage in the tournament  
26  Country playing as the home team (in the Round...  
27  Country playing as the away team  (in the Roun...  
28  TRUE whenever the host country is playing as t...  
29                    Unique identifier for the match  
30             Year in which the World Cup took place  
31                                  Data of the match  
32  Stage in the tournament (previous World Cups h...  
3

In [4]:
world_cups= pd.read_csv('world_cups.csv')
world_cups_matches = pd.read_csv('world_cup_matches.csv')
international_matches= pd.read_csv('international_matches.csv')
world_cup_squads=pd.read_csv('2022_world_cup_squads.csv',encoding= 'unicode_escape')
world_cup_matches=pd.read_csv('2022_world_cup_matches.csv')
world_cup_groups = pd.read_csv('2022_world_cup_groups.csv')

In [5]:
world_cups.shape

(22, 9)

In [6]:
world_cups.head()

Year Host Country      Winner      Runners-Up    Third      Fourth  \
0  1930      Uruguay     Uruguay       Argentina      USA  Yugoslavia   
1  1934        Italy       Italy  Czechoslovakia  Germany     Austria   
2  1938       France       Italy         Hungary   Brazil      Sweden   
3  1950       Brazil     Uruguay          Brazil   Sweden       Spain   
4  1954  Switzerland  Germany FR         Hungary  Austria     Uruguay   

   Goals Scored  Qualified Teams  Matches Played  
0          70.0               13              18  
1          70.0               16              17  
2          84.0               15              18  
3          88.0               13              22  
4         140.0               16              26

In [7]:
world_cups['Winner'].value_counts()

Brazil        5
Italy         4
Germany FR    3
Uruguay       2
Argentina     2
France        2
England       1
Spain         1
Germany       1
Name: Winner, dtype: int64

In [8]:
world_cups['Runners-Up'].value_counts()

Argentina         3
Germany FR        3
Netherlands       3
Czechoslovakia    2
Hungary           2
Brazil            2
Italy             2
Sweden            1
Germany           1
France            1
Croatia           1
Name: Runners-Up, dtype: int64

In [9]:
world_cups['Third'].value_counts()

Germany        3
Brazil         2
Sweden         2
France         2
Poland         2
USA            1
Austria        1
Chile          1
Portugal       1
Germany FR     1
Italy          1
Croatia        1
Turkey         1
Netherlands    1
Belgium        1
Name: Third, dtype: int64

In [10]:
world_cups['Fourth'].value_counts()

Uruguay           3
Yugoslavia        2
Brazil            2
England           2
Austria           1
Sweden            1
Spain             1
Germany FR        1
Soviet Union      1
Italy             1
France            1
Belgium           1
Bulgaria          1
Netherlands       1
Korea Republic    1
Portugal          1
Name: Fourth, dtype: int64

In [11]:
world_cups['Host Country'].value_counts()

Mexico          2
France          2
Brazil          2
Italy           2
Germany         2
Spain           1
Russia          1
South Africa    1
Korea/Japan     1
USA             1
Uruguay         1
Argentina       1
England         1
Chile           1
Sweden          1
Switzerland     1
Qatar           1
Name: Host Country, dtype: int64

### **From World Cups Table :**
+ **Note(1) :** Brazil Is World Cup Winner **5** times
+ **Note(2) :** Brazil Is World Cup Runners-Up **2** times
+ **Note(3) :** Brazil Is World Cup Third **2** times
+ **Note(4) :** Brazil Is World Cup Fourth **2** times
+ **Note(5) :** Brazil Is World Cup Host Country **2** times


In [12]:
world_cups_matches.shape

(900, 10)

In [13]:
world_cups_matches.head()

ID  Year        Date        Stage      Home Team  Home Goals  Away Goals  \
0   1  1930  1930-07-13  Group stage         France           4           1   
1   2  1930  1930-07-13  Group stage  United States           3           0   
2   3  1930  1930-07-14  Group stage     Yugoslavia           2           1   
3   4  1930  1930-07-14  Group stage        Romania           3           1   
4   5  1930  1930-07-15  Group stage      Argentina           1           0   

  Away Team Win Conditions  Host Team  
0    Mexico            NaN      False  
1   Belgium            NaN      False  
2    Brazil            NaN      False  
3      Peru            NaN      False  
4    France            NaN      False

In [14]:
brazil_world_cups_matches = world_cups_matches[(world_cups_matches['Home Team']=='Brazil')|(world_cups_matches['Away Team']=='Brazil')]

In [15]:
brazil_world_cups_matches

ID  Year        Date           Stage   Home Team  Home Goals  \
2      3  1930  1930-07-14     Group stage  Yugoslavia           2   
11    12  1930  1930-07-20     Group stage      Brazil           4   
23    24  1934  1934-05-27     Round of 16       Spain           3   
40    41  1938  1938-06-05     Round of 16      Brazil           6   
44    45  1938  1938-06-12  Quarter-finals      Brazil           1   
..   ...   ...         ...             ...         ...         ...   
846  847  2018  2018-06-17     Group stage      Brazil           1   
859  860  2018  2018-06-22     Group stage      Brazil           2   
878  879  2018  2018-06-27     Group stage      Serbia           0   
888  889  2018  2018-07-02     Round of 16      Brazil           2   
893  894  2018  2018-07-06  Quarter-finals      Brazil           1   

     Away Goals       Away Team Win Conditions  Host Team  
2             1          Brazil            NaN      False  
11            0         Bolivia            NaN      False  
23            1          Brazil            NaN      False  
40            5          Poland     Extra time      False  
44            1  Czechoslovakia            NaN      False  
..          ...             ...            ...        ...  
846           1     Switzerland            NaN      False  
859           0      Costa Rica            NaN      False  
878           2          Brazil            NaN      False  
888           0          Mexico            NaN      False  
893           2         Belgium            NaN      False  

[109 rows x 10 columns]

In [50]:
!pip install openpyxl

     |████████████████████████████████| 242 kB 382 kB/s eta 0:00:01


In [58]:
x = pd.DataFrame()

In [59]:
x['Stage'] = brazil_world_cups_matches['Stage'].value_counts().index
x['Times'] = brazil_world_cups_matches['Stage'].value_counts().values

In [62]:
x.to_csv('pie-data.csv',index=False)

Stage
0     43
1     14
2     11
3      9
4      8
5      8
6      6
7      4
8      3
9      3

In [17]:
brazil_home_goals = brazil_world_cups_matches[(brazil_world_cups_matches['Home Team']=='Brazil')][['Home Goals']].sum()
brazil_away_goals = brazil_world_cups_matches[(brazil_world_cups_matches['Away Team']=='Brazil')][['Away Goals']].sum()
print(brazil_home_goals)
print(brazil_away_goals)


Home Goals    176
dtype: int64
Away Goals    53
dtype: int64


In [18]:
played_against_brazil = list(brazil_world_cups_matches['Away Team'].unique())+list(brazil_world_cups_matches['Home Team'].unique())

In [19]:
played_against_brazil.remove('Brazil')

In [20]:
played_against_brazil = list(set(played_against_brazil))

In [21]:
played_against_brazil

['Denmark',
 'Poland',
 'Italy',
 'Germany',
 'Morocco',
 'Algeria',
 'United States',
 'Austria',
 'Brazil',
 'Serbia',
 'Australia',
 'Ivory Coast',
 'Bolivia',
 'England',
 'Norway',
 'Yugoslavia',
 'Japan',
 'Hungary',
 'Bulgaria',
 'Northern Ireland',
 'Turkey',
 'Wales',
 'North Korea',
 'Romania',
 'Cameroon',
 'Argentina',
 'Switzerland',
 'Portugal',
 'Croatia',
 'Ghana',
 'China',
 'Mexico',
 'Sweden',
 'Spain',
 'Chile',
 'Russia',
 'France',
 'Peru',
 'Uruguay',
 'Belgium',
 'Colombia',
 'New Zealand',
 'Czechoslovakia',
 'East Germany',
 'Scotland',
 'Netherlands',
 'Costa Rica',
 'DR Congo']

### **From World Cups Matches Table :**
+ **Note(1) :** Brazil Played **109** matchs in World Cup Over-All
+ **Note(1) :** Brazil Played **43**  matchs in World Cup Group stage
+ **Note(1) :** Brazil Played **14**  matchs in World Cup Quarter-finals
+ **Note(1) :** Brazil Played **11**  matchs in World Cup Round of 16 
+ **Note(1) :** Brazil Played **9**   matchs in World Cup First group stage
+ **Note(1) :** Brazil Played **8**   matchs in World Cup Second group stage
+ **Note(1) :** Brazil Played **8**   matchs in World Cup Semi-finals
+ **Note(1) :** Brazil Played **6**   matchs in World Cup Final
+ **Note(1) :** Brazil Played **4**   matchs in World Cup Third place
+ **Note(1) :** Brazil Played **3**   matchs in World Cup First round 
+ **Note(1) :** Brazil Played **3**   matchs in World Cup Final round
+ **Note(1) :** Brazil Scored **176**   Goal in World Cup as Home Team
+ **Note(1) :** Brazil Scored **53**    Goal in World Cup as Away Team

In [22]:
international_matches.head()

ID Tournament        Date Home Team  Home Goals  Away Goals Away Team  \
0   1   Friendly  1872-11-30  Scotland           0           0   England   
1   2   Friendly  1873-03-08   England           4           2  Scotland   
2   3   Friendly  1874-03-07  Scotland           2           1   England   
3   4   Friendly  1875-03-06   England           2           2  Scotland   
4   5   Friendly  1876-03-04  Scotland           3           0   England   

  Win Conditions  Home Stadium  
0            NaN          True  
1            NaN          True  
2            NaN          True  
3            NaN          True  
4            NaN          True

In [23]:
brazil_international_matches = international_matches[(international_matches['Home Team']=='Brazil')|(international_matches['Away Team']=='Brazil')]

In [24]:
brazil_international_matches

ID                    Tournament        Date    Home Team  \
320      321                      Friendly  1914-09-20    Argentina   
321      322                     Copa Roca  1914-09-27    Argentina   
333      334                  Copa America  1916-07-08       Brazil   
334      335                  Copa America  1916-07-10    Argentina   
336      337                  Copa America  1916-07-12       Brazil   
...      ...                           ...         ...          ...   
17628  17629  FIFA World Cup qualification  2022-03-29      Bolivia   
17659  17660                      Friendly  2022-06-02  South Korea   
17681  17682                      Friendly  2022-06-06        Japan   
17734  17735                      Friendly  2022-09-23       Brazil   
17755  17756                      Friendly  2022-09-27       Brazil   

       Home Goals  Away Goals Away Team Win Conditions  Home Stadium  
320             3           0    Brazil            NaN          True  
321             0           1    Brazil            NaN          True  
333             1           1     Chile            NaN         False  
334             1           1    Brazil            NaN          True  
336             1           2   Uruguay            NaN         False  
...           ...         ...       ...            ...           ...  
17628           0           4    Brazil            NaN          True  
17659           1           5    Brazil            NaN          True  
17681           0           1    Brazil            NaN          True  
17734           3           0     Ghana            NaN         False  
17755           5           1   Tunisia            NaN         False  

[907 rows x 9 columns]

In [25]:
brazil_home_goals_in = brazil_international_matches[(brazil_international_matches['Home Team']=='Brazil')][['Home Goals']].sum()
brazil_away_goals_in = brazil_international_matches[(brazil_international_matches['Away Team']=='Brazil')][['Away Goals']].sum()
print(brazil_home_goals_in)
print(brazil_away_goals_in)

Home Goals    1285
dtype: int64
Away Goals    714
dtype: int64


In [26]:
brazil_international_matches.head(2)

ID Tournament        Date  Home Team  Home Goals  Away Goals Away Team  \
320  321   Friendly  1914-09-20  Argentina           3           0    Brazil   
321  322  Copa Roca  1914-09-27  Argentina           0           1    Brazil   

    Win Conditions  Home Stadium  
320            NaN          True  
321            NaN          True

In [27]:
brazil_international_matches.columns
bz_in =brazil_international_matches.drop('ID',axis=1)

In [28]:
bz_in.columns

Index(['Tournament', 'Date', 'Home Team', 'Home Goals', 'Away Goals',
       'Away Team', 'Win Conditions', 'Home Stadium'],
      dtype='object')

In [29]:
bz_wc = brazil_world_cups_matches.drop(['ID','Year','Stage','Host Team'],axis=1)
bz_wc['Tournament']='World Cup'
bz_wc['Home Stadium']=False

In [30]:
bz_wc.columns

Index(['Date', 'Home Team', 'Home Goals', 'Away Goals', 'Away Team',
       'Win Conditions', 'Tournament', 'Home Stadium'],
      dtype='object')

In [31]:
brazil_matches = pd.concat([bz_in,bz_wc],axis=0)

In [32]:
brazil_matches.head(2)

Tournament        Date  Home Team  Home Goals  Away Goals Away Team  \
320   Friendly  1914-09-20  Argentina           3           0    Brazil   
321  Copa Roca  1914-09-27  Argentina           0           1    Brazil   

    Win Conditions  Home Stadium  
320            NaN          True  
321            NaN          True

In [33]:
brazil_matches.drop(['Win Conditions','Home Stadium'],axis=1,inplace=True)

In [34]:
brazil_matches.head()

Tournament        Date  Home Team  Home Goals  Away Goals Away Team
320      Friendly  1914-09-20  Argentina           3           0    Brazil
321     Copa Roca  1914-09-27  Argentina           0           1    Brazil
333  Copa America  1916-07-08     Brazil           1           1     Chile
334  Copa America  1916-07-10  Argentina           1           1    Brazil
336  Copa America  1916-07-12     Brazil           1           2   Uruguay

In [35]:
brazil_road_to_qatar = brazil_international_matches[brazil_international_matches['Tournament']=='FIFA World Cup qualification']

In [36]:
brazil_road_to_qatar.drop(['Win Conditions','Home Stadium'],axis=1,inplace=True)

/tmp/ipykernel_23332/3736564715.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  brazil_road_to_qatar.drop(['Win Conditions','Home Stadium'],axis=1,inplace=True)


In [37]:
brazil_road_to_qatar.head()

ID                    Tournament        Date Home Team  Home Goals  \
2070  2071  FIFA World Cup qualification  1954-02-28     Chile           0   
2072  2073  FIFA World Cup qualification  1954-03-07  Paraguay           0   
2074  2075  FIFA World Cup qualification  1954-03-14    Brazil           1   
2078  2079  FIFA World Cup qualification  1954-03-21    Brazil           4   
2356  2357  FIFA World Cup qualification  1957-04-13      Peru           1   

      Away Goals Away Team  
2070           2    Brazil  
2072           1    Brazil  
2074           0     Chile  
2078           1  Paraguay  
2356           1    Brazil

In [38]:
brazil_road_to_qatar.shape

(127, 7)

In [68]:
brazil_road_to_qatar.to_csv('brazil_road_to_qatar.csv',index=False)

In [63]:
brazil_matches.to_csv('brazil_matches.csv',index=False)

In [41]:
brazil_squad = world_cup_squads[world_cup_squads['Team']=='Brazil']

In [42]:
brazil_squad.drop('ID',axis=1,inplace=True)

/tmp/ipykernel_23332/2559034896.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  brazil_squad.drop('ID',axis=1,inplace=True)


In [67]:
brazil_squad[brazil_squad['Goals']==brazil_squad['Goals'].max()]

Team Position  Player  Age  Caps  Goals  WC Goals  League  \
632  Brazil  Forward  Neymar   30   121     75         6  France   

                    Club  
632  Paris Saint-Germain

In [79]:
brazil_squad.to_csv('brazil_squad.csv',index=False)

In [44]:
brazil_wc2202 = world_cup_matches[(world_cup_matches['Home Team']=='Brazil')|(world_cup_matches['Home Team']=='Brazil')]

In [45]:
brazil_wc2202.to_csv('brazil_wc_2022.csv')

In [46]:
brazil_group = world_cup_groups[world_cup_groups['Group']=='G']

In [47]:
brazil_group.to_csv('brazil_group.csv')

In [73]:
brazil_road_to_qatar['Date'].max()


'2022-03-29'

In [78]:
brazil_road_to_qatar[brazil_road_to_qatar['Date']>'2020']

ID                    Tournament        Date  Home Team  Home Goals  \
17010  17011  FIFA World Cup qualification  2020-10-09     Brazil           5   
17033  17034  FIFA World Cup qualification  2020-10-13       Peru           2   
17059  17060  FIFA World Cup qualification  2020-11-13     Brazil           1   
17083  17084  FIFA World Cup qualification  2020-11-17    Uruguay           0   
17174  17175  FIFA World Cup qualification  2021-06-04     Brazil           2   
17203  17204  FIFA World Cup qualification  2021-06-08   Paraguay           0   
17339  17340  FIFA World Cup qualification  2021-09-02      Chile           0   
17392  17393  FIFA World Cup qualification  2021-09-09     Brazil           2   
17404  17405  FIFA World Cup qualification  2021-10-07  Venezuela           1   
17427  17428  FIFA World Cup qualification  2021-10-10   Colombia           0   
17455  17456  FIFA World Cup qualification  2021-10-14     Brazil           4   
17469  17470  FIFA World Cup qualification  2021-11-11     Brazil           1   
17510  17511  FIFA World Cup qualification  2021-11-16  Argentina           0   
17563  17564  FIFA World Cup qualification  2022-01-27    Ecuador           1   
17579  17580  FIFA World Cup qualification  2022-02-01     Brazil           4   
17593  17594  FIFA World Cup qualification  2022-03-24     Brazil           4   
17628  17629  FIFA World Cup qualification  2022-03-29    Bolivia           0   

       Away Goals  Away Team  
17010           0    Bolivia  
17033           4     Brazil  
17059           0  Venezuela  
17083           2     Brazil  
17174           0    Ecuador  
17203           2     Brazil  
17339           1     Brazil  
17392           0       Peru  
17404           3     Brazil  
17427           0     Brazil  
17455           1    Uruguay  
17469           0   Colombia  
17510           0     Brazil  
17563           1     Brazil  
17579           0   Paraguay  
17593           0      Chile  
17628           4     Brazil

In [91]:
x = world_cup_squads[world_cup_squads['Team']=='Brazil'][['Age']].sum()/26
int(x)

27

In [92]:
wc_teams = list(world_cup_squads['Team'].value_counts().index)
teams_avg_age = []
teams_max_age = []
teams_min_age = []
for team in wc_teams:
    x = world_cup_squads[world_cup_squads['Team']==team][['Age']].sum()/26
    teams_avg_age.append(int(x))
    y = world_cup_squads[world_cup_squads['Team']==team][['Age']].max()
    teams_max_age.append(int(y))
    z = world_cup_squads[world_cup_squads['Team']==team][['Age']].min()
    teams_min_age.append(int(z))
        

In [94]:
new_ii = {'Team':wc_teams,
          'Min Age':teams_min_age,
          'Max Age':teams_max_age,
          'Avg Age':teams_avg_age
         }
_age = pd.DataFrame(new_ii)

In [102]:
_age

Team  Min Age  Max Age  Avg Age
0         Ecuador       20       35       25
1     Netherlands       19       39       26
2     South Korea       21       33       27
3        Portugal       19       39       26
4           Ghana       18       34       24
5     Switzerland       20       33       27
6          Serbia       21       34       26
7        Cameroon       19       33       26
8          Brazil       21       39       27
9         Morocco       18       33       26
10        Croatia       20       37       27
11         Canada       20       39       26
12        Belgium       19       35       27
13          Spain       18       34       25
14          Japan       21       39       27
15        Germany       18       36       26
16     Costa Rica       18       37       27
17        Tunisia       19       38       27
18         France       20       37       26
19        Denmark       22       36       27
20      Australia       18       37       27
21   Saudi Arabia       21       33       27
22         Poland       20       35       27
23         Mexico       23       40       28
24      Argentina       21       36       27
25          Wales       20       35       26
26  United States       19       35       25
27        England       19       32       26
28        Senegal       20       33       26
29          Qatar       20       32       26
30        Uruguay       20       36       27
31           Iran       24       34       27

In [103]:
world_cup_groups

Group           Team  FIFA Ranking
0      A          Qatar            50
1      A        Ecuador            44
2      A        Senegal            18
3      A    Netherlands             8
4      B        England             5
5      B           Iran            20
6      B  United States            16
7      B          Wales            19
8      C      Argentina             3
9      C   Saudi Arabia            51
10     C         Mexico            13
11     C         Poland            26
12     D         France             4
13     D      Australia            38
14     D        Denmark            10
15     D        Tunisia            30
16     E          Spain             7
17     E     Costa Rica            31
18     E        Germany            11
19     E          Japan            24
20     F        Belgium             2
21     F         Canada            41
22     F        Morocco            22
23     F        Croatia            12
24     G         Brazil             1
25     G         Serbia            21
26     G    Switzerland            15
27     G       Cameroon            43
28     H       Portugal             9
29     H          Ghana            61
30     H        Uruguay            14
31     H    South Korea            28

In [113]:
world_cup_groups[world_cup_groups['Group']=='G']

Group         Team  FIFA Ranking
24     G       Brazil             1
25     G       Serbia            21
26     G  Switzerland            15
27     G     Cameroon            43

In [116]:
print(_age[_age['Team']=='Brazil'])
print('---------------------------')
print(_age[_age['Team']=='Serbia'])
print('---------------------------')
print(_age[_age['Team']=='Switzerland'])
print('---------------------------')
print(_age[_age['Team']=='Cameroon'])
print('---------------------------')


     Team  Min Age  Max Age  Avg Age
8  Brazil       21       39       27
---------------------------
     Team  Min Age  Max Age  Avg Age
6  Serbia       21       34       26
---------------------------
          Team  Min Age  Max Age  Avg Age
5  Switzerland       20       33       27
---------------------------
       Team  Min Age  Max Age  Avg Age
7  Cameroon       19       33       26
---------------------------
